frontend/src/metabase/forms/components/FormObserver/FormObserver.tsx

```diff
@@ -0,0 +1,22 @@
+import { useFormikContext } from "formik";
+import { useEffect } from "react";
+
+interface FormObserverProps<T> {
+  onChange: (vals: T) => void;
+}
+
+// This component can be used to effectivy add an onChange handler to a from.
+// however, this should be used with caution as it is bad practice to duplicate
+// state.
```

Comment:
👍

Reply:
I think if you use [ jsdoc style](https://jsdoc.app/about-getting-started#:~:text=/**%20This%20is%20a%20description%20of%20the%20foo%20function.%20*/) comment this will play nicer with editors

Reply:
Fair point. I'll update haha